internal/subshell/subshell.go

```diff
@@ -142,3 +134,35 @@ func New() SubShell {
 
 	return subs
 }
+
+func DetectShellBinary(cfg *config.Instance) (binary string) {
+	defer func() {
```

Comment:
This is going to cause config writes even when we received the shell from the config in the first place.